src/main/java/qtlog/LogParser/LogParser.java

```diff
@@ -0,0 +1,257 @@
+package qtlog.LogParser;
+
+import qtlog.shared.FileDTO;
+import qtlog.shared.RawLogDTO;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class LogParser implements ILogParser{
+    private int agentCount;
+    private int skillCount;
+    @Override
+    public RawLogDTO giveLogToParse(FileDTO fileInfo) {
+        byte[] fileContent;
+        ByteArrayInputStream byteArrayStream;
+        Path filePath = Path.of(fileInfo.getFilepath());
+
+        if(logIsZipped(fileInfo.getFilename())){
+            try {
+                byte[] unzippedLog = unzipLog(fileInfo.getFilepath());
+                byteArrayStream = new ByteArrayInputStream(unzippedLog);
+                return parseLog(byteArrayStream);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            try {
+                fileContent = Files.readAllBytes(filePath);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```

Comment:
Is there a particular reason why you put a try/catch statement under each branch of the if/else instead of just wrapping the whole if? They are both only catching IOExceptions anyway.